File: app/src/main/java/com/m1guelgtz/templatecarsapi/Demo/Features/Tasks/Domain/Entities/Task.kt
package com.m1guelgtz.templatecarsapi.Demo.Features.Tasks.Domain.Entities

data class Task(
    val id: String,
    val projectId: String,
    val title: String,
    val description: String?,
    val status: TaskStatus,
    val assignedTo: String?,
    val createdBy: String,
    val version: Int,
    val updatedAt: String,
    val createdAt: String
)

enum class TaskStatus {
    TODO, IN_PROGRESS, REVIEW, DONE
}

File: app/src/main/java/com/m1guelgtz/templatecarsapi/src_api/domain/entities/task.py
from dataclasses import dataclass
from datetime import datetime
from uuid import UUID


@dataclass
class Task:
    id: UUID
    project_id: UUID
    title: str
    description: str | None
    status: str  # 'TODO' | 'IN_PROGRESS' | 'REVIEW' | 'DONE'
    assigned_to: UUID | None
    created_by: UUID
    version: int
    updated_at: datetime
    created_at: datetime
